src/main/java/me/philippheuer/twitch4j/message/irc/listener/IRCEventListener.java

```diff
@@ -213,23 +212,61 @@ public void onChannnelClientLeaveEvent(IRCMessageEvent event) {
 	public void onChannelModChange(IRCMessageEvent event) {
 		if(event.getCommandType().equals("MODE") && event.getPayload().isPresent()) {
 			// Recieving Mod Status
-			if(event.getPayload().get().startsWith("+o")) {
+			if(event.getPayload().get().substring(1).startsWith("o")) {
 				// Load Info
-				Channel channel = event.getClient().getChannelEndpoint(event.getChannel().get()).getChannel();
+				Channel channel = event.getClient().getChannelEndpoint(event.getChannelName().get()).getChannel();
 				User user = event.getClient().getUserEndpoint().getUserByUserName(event.getPayload().get().substring(3)).get();
 
 				// Dispatch Event
-				event.getClient().getDispatcher().dispatch(new ChannelModEvent(channel, user, true));
+				event.getClient().getDispatcher().dispatch(new ChannelModEvent(channel, user, event.getPayload().get().startsWith("+")));
 			}
-			// Losing Mod Status
-			else if(event.getPayload().get().startsWith("-o")) {
-				// Load Info
-				Channel channel = event.getClient().getChannelEndpoint(event.getChannel().get()).getChannel();
-				User user = event.getClient().getUserEndpoint().getUserByUserName(event.getPayload().get().substring(3)).get();
+		}
+	}
 
-				// Dispatch Event
-				event.getClient().getDispatcher().dispatch(new ChannelModEvent(channel, user, false));
+	@EventSubscriber
+	public void onNoticeEvent(IRCMessageEvent event) {
+		if (event.getCommandType().equals("NOTICE")) {
+			Channel channel = event.getClient().getChannelEndpoint(event.getChannelName().get()).getChannel();
+			String messageId = event.getTagValue("msg-id").get();
+			String message = event.getMessage().get();
+
+			event.getClient().getDispatcher().dispatch(new NoticeEvent(channel, messageId, message));
+		}
+	}
+
+	@EventSubscriber
+	public void onChannelState(IRCMessageEvent event) {
+		if(event.getCommandType().equals("ROOMSTATE")) {
+			// getting Status on channel
+			Channel channel = event.getClient().getChannelEndpoint(event.getChannelId()).getChannel();
+			Map<ChannelStateEvent.ChannelState, Object> states = new HashMap<ChannelStateEvent.ChannelState, Object>();
+			if (event.getTags().size() > 2) {
+				event.getTags().forEach((k, v) -> {
+					switch (k) {
+						case "broadcaster-lang":
+							states.put(ChannelStateEvent.ChannelState.BROADCAST_LANG, Locale.forLanguageTag(v));
```

Comment:
That's the issue for [this](https://github.com/PhilippHeuer/twitch4j/pull/20#issuecomment-330082243).